test/e2e/logforwarding/forward_to_fluent_test.go

```diff
@@ -0,0 +1,155 @@
+package logforwarding
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	logforward "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
+	"github.com/openshift/cluster-logging-operator/pkg/logger"
+	"github.com/openshift/cluster-logging-operator/test/helpers"
+	apps "k8s.io/api/apps/v1"
+)
+
+var _ = Describe("LogForwarding", func() {
+	_, filename, _, _ := runtime.Caller(0)
+	logger.Infof("Running %s", filename)
+	var (
+		err              error
+		fluentDeployment *apps.Deployment
+		e2e              = helpers.NewE2ETestFramework()
+		rootDir          string
+	)
+	BeforeEach(func() {
+		e2e.DeployLogGenerator()
+		rootDir = filepath.Join(filepath.Dir(filename), "..", "..", "..", "/")
+	})
+	Describe("when ClusterLogging is configured with 'forwarding' to an administrator managed fluentd", func() {
+
+		Context("and the receiver is unsecured", func() {
+
+			BeforeEach(func() {
+				if fluentDeployment, err = e2e.DeployFluendReceiver(rootDir, false); err != nil {
+					Fail(fmt.Sprintf("Unable to deploy fluent receiver: %v", err))
+				}
+
+				cr := helpers.NewClusterLogging(helpers.ComponentTypeCollector)
+				if err := e2e.CreateClusterLogging(cr); err != nil {
+					Fail(fmt.Sprintf("Unable to create an instance of cluster logging: %v", err))
+				}
+				forwarding := &logforward.LogForwarding{
```

Comment:
This is missing TypeMeta: metav1.TypeMeta ; which defines the Kind of resource and APIVersion

Reply:
> This is missing TypeMeta: metav1.TypeMeta ; which defines the Kind of resource and APIVersion

While this is true, the framework code actually serializes the struct and makes a straight REST call to the api server which must rely on the URI to determine the type instead of the struct.

Reply:
Retract this statement and added a change